### ISSUE TITLE
inline image display is broken (iterm2)

### ISSUE THREAD

**Issue by y-lu** [ranger/ranger#850]
#### ISSUE TYPE

- Bug report

#### RUNTIME ENVIRONMENT
- Operating system and version: Linux 3.10.0-327.36.1.el7.x86_64 #1 SMP Wed Aug 17 03:02:37 EDT 2016 x86_64 Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz GenuineIntel GNU/Linux
- Terminal emulator and version: iTerm2 (v3.0.14)
- Python version: 3.5.2 |Anaconda 4.1.1 (64-bit)| (default, Jul  2 2016, 17:53:06) [GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
- Ranger version/commit: ranger-master 1.9.0b5
- Locale: en_US.UTF-8

#### EXPECTED BEHAVIOR
Ranger should display image previews under iTerm2.

#### CURRENT BEHAVIOR
A generic icon is displayed instead.

#### CONTEXT
The feature was working correctly at commit 10d7174613e9865b065c52e57af8b0799cd106b0, but failed as of commit 305537963bb1db5b3e50213c46749f62bb84bc9a.
It is likely some recent changes broke it.

#### STEPS TO REPRODUCE

1. enable inline image display and set method to iterm2 (by editing rc.conf accordingly)
2. launch ranger in iTerm2
3. go to an image file

#### TRACEBACK

```

```

**Comment by Vifon** [comment]
How exactly are you running iTerm2 on Linux? Are you running ranger via SSH?

**Comment by y-lu** [comment]
Hi,

I was running iTerm2 on Mac, and ranger via SSH.

Thanks,

-Yong

**Comment by randynobx** [comment]
@y-lu Could you:

1) Run ranger locally on the Mac (no ssh from a linux box) and see if the iTerm2 preview works then?
2) Using the same ssh setup from linux to mac, use a different preview option (w3m would probably be your best bet) and see if the preview works?

**Comment by hollandan** [comment]
I ran into this issue as well. 

I think ranger needs to know you're using a full color terminal.

In bash, you can do that by setting an environment variable like this:

`export TERM=xterm-256color`

After I did that, image previews started working for me

**Comment by tamasgal** [comment]
I have the same issue, I only see a scaled generic image. The `TERM` variable was already set to `xterm-256color`.

**Comment by toonn** [comment]
@y-lu, there's 177 commits between those... It would be great if you could do a bisect because that's too many to go through.

```bash
git clone https://github.com/ranger/ranger.git # or git pull if you already have the repo
# first test whether the issue is still present
git bisect start
git bisect bad # mark the current commit as bad, this is probably way after 3055379 but bisect does a binary search so it shouldn't make a big difference
git bisect good 10d7174 # mark the known good commit, now bisect will select a commit for you to test, test it by running `./ranger.py`
git bisect good/bad # depending on whether it worked or not, bisect will select a new commit until you narrow it down to the culprit, report this commit
git bisect reset # stop the bisect session and return to the commit you were at before it started
```

If anyone else has the issue they could of course also try bisecting.

**Comment by tamasgal** [comment]
The latest master branch (head at c8613cf8a83c9fc516cb1d0c523bf63dc79f1a70) works fine for me!

**Comment by toonn** [comment]
@tamasgal, thanks for the report. Pip's behind on master of course, I'll leave this open until we release 1.9.2, which should be soon™, so we can be sure it's not a problem with packaging.

**Comment by tamasgal** [comment]
@toonn yes, that’s why I forced the git master branch when using pip. For me it seems to be a packaging thing.

**Comment by toonn** [comment]
Oh, I didn't realize. Hmm, that's unfortunate. Tbh I'm not familiar enough with the packaging methods to figure out the problem probably.
